### PR TITLE
chore: connect achievements and badges api

### PR DESCRIPTION
**Pull request by ice-hector**
- connect achievements API;
- finalize profile badges summary UI;
- finalize my badges screen;
- add loading states for badges;
- optimized badge images;

### CODE REVIEW

File: src/store/modules/Contacts/sagas/inviteContactSaga.ts [hunk 1]
```diff
@@ -26,12 +20,9 @@ export function* inviteContactSaga(
     if (!contact.phoneNumbers.length) {
       throw new Error('Contact has no phone numbers');
     }
-
-    const url = `${LINKS.MAIN}@${username}`;
-
     const text = `${t('team.contacts_list.invitation_text', {
       name: getContactName(contact),
-    })} ${url}`;
```
Comment (ice-atlas) [inline src/store/modules/Contacts/sagas/inviteContactSaga.ts]: Why you removed it? I think `url` is part of invitation

File: src/store/modules/Achievements/sagas/loadUserAchievements.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    );
+
+    const getBadgesResult: SagaReturnType<
+      typeof Api.achievements.getBadgeSummaries
+    > = yield call(Api.achievements.getBadgeSummaries, {
+      userId,
+    });
+
+    yield put(
+      AchievementsActions.USER_ACHIEVEMENTS_LOAD.SUCCESS.create({
```
Comment (ice-atlas) [inline src/store/modules/Achievements/sagas/loadUserAchievements.ts:26]: We should have only 1 `SUCCESS` action in thread.

File: src/store/modules/Achievements/sagas/loadUserAchievements.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  action: ReturnType<typeof actionCreator>,
+) {
+  const {userId} = action.payload;
+
+  try {
+    const {
+      level,
+      roles,
+    }: SagaReturnType<typeof Api.achievements.getLevelsAndRoles> = yield call(
+      Api.achievements.getLevelsAndRoles,
```
Comment (ice-atlas) [inline src/store/modules/Achievements/sagas/loadUserAchievements.ts]: We have another saga (`loadLevelsAndRolesSaga`) for this. Adjust it pls

File: src/services/getstream/mockData.ts [hunk 4]
```diff
@@ -34,7 +34,7 @@ export const mockedNotifications = [
     verb: 'joined',
     foreign_id: 'joined:id:1',
     extra: {
-      description: '**@John** and **3 others** have joined your team',
+      description: '**@Robert** and **3 others** have joined your team',
```
Comment (ice-atlas) [inline src/services/getstream/mockData.ts]: Let's keep it as is

File: src/services/getstream/mockData.ts [hunk 5]
```diff
@@ -49,7 +49,7 @@ export const mockedNotifications = [
     verb: 'joined',
     foreign_id: 'joined:id:2',
     extra: {
-      description: '**@Doe** has joined your team',
+      description: '**@oleh** has joined your team',
```
Comment (ice-atlas) [inline src/services/getstream/mockData.ts]: same here

File: src/screens/Team/components/TierList/components/Header/components/Earnings.tsx [hunk 6]
```diff
@@ -23,40 +22,20 @@ export const Earnings = forwardRef(
         ? t('team.tier_one.header_list.title_earnings')
         : t('team.tier_two.header_list.title_earnings');
     return (
-      <View style={styles.container} ref={forwardedRef}>
-        <Text style={styles.label}>{`${title}:`}</Text>
-
-        <FormattedNumber
```
Comment (ice-atlas) [inline src/screens/Team/components/TierList/components/Header/components/Earnings.tsx]: We need to keep this functionality

File: src/screens/SettingsFlow/PersonalInformation/index.tsx [hunk 7]
```diff
@@ -113,7 +113,6 @@ export const PersonalInformation = memo(() => {
             containerStyle={styles.input}
             icon={<EmailIcon width={rem(24)} height={rem(14)} />}
             onChange={onEmailPress}
-            numberOfLines={1}
```
Comment (ice-atlas) [inline src/screens/SettingsFlow/PersonalInformation/index.tsx]: why you decide to do so?

File: src/screens/ProfileFlow/Profile/components/Badges/components/BadgeSummariesList.tsx [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+}: Props) => {
+  const isPrivacyInfoShown = useSelector(isPrivacyInfoShownSelector);
+  const hidden = isOwner
+    ? user?.hiddenProfileElements?.includes('badges') && !isPrivacyInfoShown
+    : user?.hiddenProfileElements?.includes('badges');
+
+  const showPlaceholders =
+    !isOwner && user?.hiddenProfileElements?.includes('badges');
+
+  const badgesData = loading
```
Comment (ice-atlas) [inline src/screens/ProfileFlow/Profile/components/Badges/components/BadgeSummariesList.tsx]: let's use `useMemo` for this

File: src/screens/HomeFlow/Home/components/Pager/components/Wallet/components/TotalMiningRateValue/index.tsx [hunk 9]
```diff
@@ -30,17 +30,16 @@ export const TotalMiningRateValue = ({style}: Props) => {
   );
 
   const rateValueTextStyle = useMemo(() => {
-    switch (miningRates?.type) {
```
Comment (ice-atlas) [inline src/screens/HomeFlow/Home/components/Pager/components/Wallet/components/TotalMiningRateValue/index.tsx]: revert it pls

File: src/screens/HomeFlow/Home/components/Pager/components/MiningRate/index.tsx [hunk 10]
```diff
@@ -23,19 +23,6 @@ import {MiningRateValue} from './components/MiningRateValue';
 export const MiningRate = memo(() => {
   const miningRates = useSelector(miningRatesSelector);
 
-  const rateValueTextStyle = useMemo(() => {
```
Comment (ice-atlas) [inline src/screens/HomeFlow/Home/components/Pager/components/MiningRate/index.tsx]: revert it pls

File: src/screens/HomeFlow/Home/components/Pager/components/MiningRate/components/MiningRateValue/index.tsx [hunk 11]
```diff
@@ -11,15 +11,14 @@ interface Props {
...
-  signRequired?: boolean;
 }
 
 export const MiningRateValue = memo(
-  ({style, bodyStyle, decimalsStyle, value, signRequired = false}: Props) => {
+  ({style, bodyStyle, decimalsStyle, value}: Props) => {
     const animatedValue = useAnimatedNumber(
       value,
       initialValue =>
-        `${initialValue > 0 && signRequired ? '+' : ''}${formatNumberString(
```
Comment (ice-atlas) [inline src/screens/HomeFlow/Home/components/Pager/components/MiningRate/components/MiningRateValue/index.tsx]: revert it pls

File: src/constants/links.ts [hunk 12]
```diff
@@ -15,7 +15,7 @@ export const LINKS = {
   KNOWLEDGE_BASE: 'https://ice.io/knowledge-base',
   BONUSES: 'https://ice.io/bonuses',
   TEAM: 'https://ice.io/team',
-  FEEDBACK_EMAIL: 'feedback@ice.io',
```
Comment (ice-atlas) [inline src/constants/links.ts]: revert it pls

File: src/constants/countries.ts [hunk 13]
```diff
@@ -177,7 +177,7 @@ export const countries: Country[] = [
     iddCode: '+673',
   },
   {
-    name: 'Bolivia',
```
Comment (ice-atlas) [inline src/constants/countries.ts]: revert country names

File: src/components/Inputs/CommonInput/index.tsx [hunk 14]
```diff
@@ -71,9 +71,7 @@ export const CommonInput = ({
           <View style={styles.inputWrapper}>
             {prefix}
             {onChange ? (
-              <Text style={styles.input} {...textInputProps}>
```
Comment (ice-atlas) [inline src/components/Inputs/CommonInput/index.tsx]: revert it pls

**Review by ice-atlas — CHANGES_REQUESTED** [review]

File: src/screens/ProfileFlow/Badges/components/BadgeCard.tsx [hunk 15]
```diff
@@ -1,32 +1,90 @@
...
+  name,
+  index,
+  type,
+  achieved,
+  percentageOfUsersInProgress,
+  achievingRange,
   connector = {},
 }: Props) => {
+  const imagePath =
+    `${type}${index}_achieved_${achieved}` as keyof typeof Images.badges;
```
Comment (ice-atlas) [inline src/screens/ProfileFlow/Badges/components/BadgeCard.tsx:45]: use `as const` or put it in the one specific place as is

File: src/assets/svg/PeopleIcon.tsx [hunk 16]
```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import * as React from 'react';
+import Svg, {Mask, Path, SvgProps} from 'react-native-svg';
+
+export const PeopleIcon = (props: SvgProps) => (
+  <Svg width={13} height={13} fill="none" viewBox="0 0 13 13" {...props}>
```
Comment (ice-orion) [inline src/assets/svg/PeopleIcon.tsx]: rem and color?

File: src/constants/colors.ts [hunk 17]
```diff
@@ -42,6 +42,7 @@ export const COLORS = {
   headerBackground: 'rgba(255, 255, 255, .96)',
   aliceBlue: '#EFF9FF',
   linkWater: '#F1F4FC',
+  linkWater2: '#D7EAF4', // inactive badges color
```
Comment (ice-orion) [inline src/constants/colors.ts]: kinda weird naming, let's name it `azureishWhite`?  https://www.color-name.com/hex/d7eaf4
And comment here looks redundant

File: src/screens/ProfileFlow/Profile/components/AvatarHeader/index.tsx
Comment (ice-orion) [inline src/screens/ProfileFlow/Profile/components/AvatarHeader/index.tsx:1]: The component is almost 400 lines of code that is too much.
Lest separate code according to single responsibility principle:
e.g. : set all the animation related code to `useAnimatedStyles` hook
e.g. 2: create separate components for buttons - PenButton / ContactsButton
...

File: src/screens/ProfileFlow/Profile/components/Badges/components/BadgeCard.tsx [hunk 18]
```diff
@@ -48,17 +56,42 @@ export const BadgeCard = memo(
       } else {
         navigation.navigate('Badges', {category});
       }
-    }, [category, navigation, onUpdate, isProfilePrivacyEditMode]);
+    }, [category, isProfilePrivacyEditMode, onUpdate, navigation]);
+
+    const categoryTranslation = t(`profile.badge_types.${category}.title`);
+
+    const image = `${category}0_achieved_true` as keyof typeof Images.badges;
```
Comment (ice-orion) [inline src/screens/ProfileFlow/Profile/components/Badges/components/BadgeCard.tsx]: better use `as const` here as more safe

File: src/screens/ProfileFlow/Profile/components/Badges/components/BadgeCard.tsx [hunk 19]
```diff
@@ -48,17 +56,42 @@ export const BadgeCard = memo(
...
+    const inactiveImage = isPlaceholder
+      ? (`placeholder${index}` as keyof typeof Images.badges)
+      : (`${category}0_achieved_false` as keyof typeof Images.badges);
```
Comment (ice-orion) [inline src/screens/ProfileFlow/Profile/components/Badges/components/BadgeCard.tsx]: and here

File: src/screens/ProfileFlow/Profile/components/Badges/components/BadgeSummariesList.tsx [hunk 20]
```diff
@@ -0,0 +1,109 @@
...
+  loading,
+  data,
+  isProfilePrivacyEditMode,
+  user,
+  isOwner,
+}: Props) => {
+  const isPrivacyInfoShown = useSelector(isPrivacyInfoShownSelector);
+  const hidden = isOwner
+    ? user?.hiddenProfileElements?.includes('badges') && !isPrivacyInfoShown
+    : user?.hiddenProfileElements?.includes('badges');
```
Comment (ice-orion) [inline src/screens/ProfileFlow/Profile/components/Badges/components/BadgeSummariesList.tsx]: Let's use separate variable for `user?.hiddenProfileElements?.includes('badges')`?

File: src/screens/ProfileFlow/Profile/components/LadderBar/index.tsx [hunk 21]
```diff
@@ -17,35 +22,57 @@ type Props = {
...
+              ? hiddenElements.includes('referralCount') && !isPrivacyInfoShown
+              : hiddenElements.includes('referralCount')
+          }
         />
         <LadderItem
           title={t('global.level').toUpperCase()}
-          value={'21'}
+          value={
+            achievements?.levelsAndRoles?.level
+              ? `${achievements?.levelsAndRoles?.level}`
```
Comment (ice-orion) [inline src/screens/ProfileFlow/Profile/components/LadderBar/index.tsx:65]: Also, can't the level be 0? In this case the ternar will return wrong result
optional: `achievements.levelsAndRoles.level`

Comment (ice-hector) [reply src/screens/ProfileFlow/Profile/components/LadderBar/index.tsx:65]: 1 is default, not less

File: src/screens/ProfileFlow/Badges/components/BadgeCard.tsx (hunk 15 above)
Comment (ice-hector) [reply src/screens/ProfileFlow/Badges/components/BadgeCard.tsx:45]: Index here is a number so couldn't be use `as const`
This solution is the temp in general, so we gonna move that images to the backend side and use only URLs on FE side after release.

File: src/store/modules/Achievements/sagas/loadAllBadgesSaga.ts [hunk 22]
```diff
@@ -0,0 +1,51 @@
...
+    });
+
+    const getBadgesByTypeResultLevel: SagaReturnType<
+      typeof Api.achievements.getBadgesByType
+    > = yield call(Api.achievements.getBadgesByType, {
+      type: 'level',
+      userId,
+    });
+
+    const getBadgesByTypeResultSocial: SagaReturnType<
```
Comment (ice-orion) [inline src/store/modules/Achievements/sagas/loadAllBadgesSaga.ts]: can we do getBadgesByType getBadgesByType and getBadgesByType in parallel?

File: src/screens/ProfileFlow/Profile/components/Badges/components/BadgeSummariesList.tsx [hunk 23]
```diff
@@ -0,0 +1,111 @@
...
+  const hidden = isOwner
+    ? areBadgesHidden && !isPrivacyInfoShown
+    : areBadgesHidden;
+
+  const defaultData =
+    data.length === 0 ? Array(NUMBER_OF_SKELETONS).fill(null) : data;
+
+  const badgesData = useMemo(() => {
+    return loading
+      ? defaultData
```
Comment (ice-atlas) [inline src/screens/ProfileFlow/Profile/components/Badges/components/BadgeSummariesList.tsx]: basic idea to use `memo` here also simplify (get rid of) ternary operator. Pls use `if-else` here

File: src/store/modules/Achievements/sagas/loadAllBadgesSaga.ts [hunk 24]
```diff
@@ -0,0 +1,59 @@
...
+    >;
+    let getBadgesByTypeResultSocial: SagaReturnType<
+      typeof Api.achievements.getBadgesByType
+    >;
+
+    [
+      getBadgesByTypeResultCoin,
+      getBadgesByTypeResultLevel,
+      getBadgesByTypeResultSocial,
+    ] = yield all([
```
Comment (ice-atlas) [inline src/store/modules/Achievements/sagas/loadAllBadgesSaga.ts]: Let's try to use `TypeScript tuples` for this

**Review by ice-atlas — CHANGES_REQUESTED** [review]

**Review by ice-orion — APPROVED** [review]

**Review by ice-hades — APPROVED** [review]

**Review by ice-atlas — APPROVED** [review]

**Pull request merged**